server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -751,6 +764,16 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
         })
     );
 
+    public <T> void writeListGeneric(List<T> v, Writer<T> writer) throws IOException {
+        writeByte((byte) 7);
+        writeCollection(v, writer);
```

Comment:
Why is a writer needed as a parameter, for a generic list it should always be writeGenericValue?

Reply:
I set it up this way to so I can pass `writeStringGeneric` for the string case and save the type lookup there.
It's a bit of a strange method admittedly, but this is already a bit of a special case and I figured I'd try to keep the code dry with the magic `7` and all here.